arch/arm64/core/CMakeLists.txt

```diff
@@ -53,4 +53,6 @@ if(CMAKE_C_COMPILER_ID STREQUAL "GNU")
   zephyr_cc_option(-moverride=tune=no_ldp_stp_qregs)
 endif()
 
-add_subdirectory_ifdef(CONFIG_SOC_XENVM xen)
+if (CONFIG_SOC_XENVM OR CONFIG_DT_HAS_XEN_XEN_4_15_ENABLED)
```

Comment:
Ugh. Why do you hardcoding Xen version there?  

For this exact reason, there are two separate compatible entries: `compatible = "xen,xen-4.15", "xen,xen";` please use the second one if there is no hard dependency on this particular Xen version. 

Also, please take into account that 4.15 is not an actual Xen version running on a platform.

Reply:
because CONFIG_DT_HAS_XEN_XEN_ENABLED was defined during build and GitHub automation notified about that
if you want I can add extra check here too

Reply:
Probably Zephyr DT parser does not add all compatibles to these defines. I think it will be okay to remove 4.15 from DT compatibles at all. All Zephyr drivers uses `xen,xen`. I don't think that we should update exact version every time.